plugins/crypto.py

```diff
@@ -3,24 +3,107 @@
 from util import http, hook
 
 
-@hook.command()
-def crypto(inp, say=None):
-    fsym = inp.split(' ')[0].upper()
-    price_response = http.get_json("https://min-api.cryptocompare.com/data/pricemultifull", fsyms=fsym, tsyms="USD")
+API_FIELD_TRANSLATION = {
+    'FROMSYMBOL': 'coin',
+    'TOSYMBOL': 'currency',
+    'PRICE': 'last',
+    'HIGH24HOUR': 'high',
+    'LOW24HOUR': 'low',
+    'MKTCAP': 'market_cap_currency',
+    'TOTALVOLUME24H': 'volume_coin',
+    'TOTALVOLUME24HTO': 'volume_currency',
+    'SUPPLY': 'supply_currency',
+}
 
-    if price_response.get("Response") == "Error":
-        return "Error: " + price_response.get("Message", "An API Error Has Occured. https://min-api.cryptocompare.com/")
 
-    if not price_response["DISPLAY"].get(fsym):
-        return "Error: Unable to lookup pricing for {}".format(fsym)
+CURRENCY_SYMBOLS = {
+    'USD': u'$',
+    'CAD': u'$',
+    'ETH': u'\u039E',
+    'BTC': u'\u20BF',
+    'RUB': u'\u20BD',
+    'JPY': u'\u00A5',
+    'CNY': u'\u00A5',
+    'EUR': u'\u20AC',
+    'AZN': u'\u20BC',
+    'ILS': u'\u20AA',
+    'THB': u'\u0E3F',
+    'KRW': u'\u20A9',
+    'KPW': u'\u20A9',
+    'VND': u'\u20AB',
+    'SEK': u'kr',
+    'ZAR': u'R'
+}
 
-    values = price_response["DISPLAY"][fsym]["USD"]
 
-    say(
-        (
-            "USD/{FROMSYMBOL}: \x0307{PRICE}\x0f - High: \x0307{HIGHDAY}\x0f "
-            "- Low: \x0307{LOWDAY}\x0f - Volume: {VOLUMEDAY}"
-            " ({VOLUMEDAYTO}) - Total Supply: {SUPPLY} - MktCap: {MKTCAP}"
-        ).format(**values)
+def get_coin_price(coin, price_basis):
+    """
+    Get the coin price against a basis price system, eg BTC to USD.
+
+    :param coin: the cryptocurrency currency to check
+    :param price_basis: the price basis to check against
+    :return: a dictionary which includes the coin price, symbols,
+             and other stats, or none if something doesn't exist
+    :rtype: dict or none
+    """
+    coin = coin.upper()
+    price_basis = price_basis.upper()
+
+    response = http.get_json(
+        'https://min-api.cryptocompare.com/data/pricemultifull',
+        query_params={
+            'fsyms': coin,
+            'tsyms': price_basis
+        }
     )
 
+    if not response or 'RAW' not in response:
+        return None
+
+    raw_coin_price = response['RAW'][coin.upper()][price_basis.upper()]
+
+    coin_price = {API_FIELD_TRANSLATION[k]: v for k, v in raw_coin_price.items() if k in API_FIELD_TRANSLATION}
+
+    if coin_price['currency'] in CURRENCY_SYMBOLS:
+        coin_price['currency_symbol'] = CURRENCY_SYMBOLS[coin_price['currency']]
+    else:
+        coin_price['currency_symbol'] = '%s ' % coin_price['currency']
+
+    return coin_price
+
+
+@hook.command(autohelp=False)
```

Comment:
having this default to 'BTC USD' is silly when there's a dedicated `.bit`

Reply:
You think it should just fail?

The default for the original was to use the price basis of USD but to fail with `Error: fsyms param is empty or null.` which isn't super helpful.